internal/uvm/network.go

```diff
@@ -119,12 +121,16 @@ func GetNamespaceEndpoints(ctx context.Context, netNS string) ([]*hns.HNSEndpoin
 
 // NCProxyEnabled returns if there is a network configuration client.
 func (uvm *UtilityVM) NCProxyEnabled() bool {
-	return uvm.ncProxyClient != nil
+	return uvm.ncProxyClientAddress != ""
 }
 
-// NCProxyClient returns the network configuration proxy client.
-func (uvm *UtilityVM) NCProxyClient() ncproxyttrpc.NetworkConfigProxyService {
-	return uvm.ncProxyClient
+func (uvm *UtilityVM) GetNCProxyClient() (ncproxyttrpc.NetworkConfigProxyService, error) {
+	conn, err := winio.DialPipe(uvm.ncProxyClientAddress, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to ncproxy service")
+	}
+	client := ttrpc.NewClient(conn, ttrpc.WithOnClose(func() { conn.Close() }))
```

Comment:
How does the underlying `conn` get cleaned up? I see we will close it when the client is closed here, but then we cast it to a type which doesn't implement `io.Closer`.

Reply:
Ping on this

Reply:
You're correct, it gets leaked from reading through the code. I'll work on a fix for this.

Reply:
I've fixed this in recent iterations

Reply:
Keeping this open so this can be reviewed and validated as such :)